stdlib/src/utils/static_tuple.mojo

```diff
@@ -369,27 +369,26 @@ struct InlineArray[ElementType: CollectionElement, size: Int](Sized):
 
     @always_inline("nodebug")
     fn __refitem__[
-        IntableType: Intable,
-    ](self: Reference[Self, _, _], index: IntableType) -> Reference[
+        IndexerType: Indexer,
+    ](self: Reference[Self, _, _], index: IndexerType) -> Reference[
         Self.ElementType, self.is_mutable, self.lifetime
     ]:
         """Get a `Reference` to the element at the given index.
 
         Parameters:
-            IntableType: The inferred type of an intable argument.
+            IndexerType: The inferred type of an indexer argument.
 
         Args:
             index: The index of the item.
 
         Returns:
             A reference to the item at the given index.
         """
-        debug_assert(-size <= int(index) < size, "Index must be within bounds.")
-        var normalized_idx = int(index)
-        if normalized_idx < 0:
-            normalized_idx += size
+        var normalized_index = normalize_index[container_name="InlineArray"](
```

Comment:
This is subtly different behavior: `debug_assert` can be disabled (in which case there is no bound checking, no overhead, but bad things can happen); the implementation you introduced, however, always performs the bounds check, which guarantees that this becomes more expensive.

@JoeLoser Any thoughts on whether this should be a "safe" method or not?

Reply:
Thanks for the review. This is intentional. I followed the instructions from @JoeLoser  here: https://github.com/modularml/mojo/pull/2386#discussion_r1589103000 

Since having a performant access to the elements in any array is desirable, we could get some inspiration from rust which does have the same challenge. Rust solves it by having safe access with `[i]` and unsafe access with [`get_unchecked`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get_unchecked) . 

What I would recommend here is to do something similar, as unsafe things must be explicitly unsafe using an appropriate name (from the stdlib vision document). When this PR gets merged, I can open an issue (which can be a good first issue!) to add a second method to the InlineArray `unsafe_get()` where `normalized_index` isn't called, and we don't wrap the index when there are negative values. A debug_assert will be added in `unsafe_get()` for better debuggability.

I'll answer here concerning the heap memory allocation. I think there are a few points here to add:
* The heap allocation will be used only when the "safe" get is used. Thus if the platform doesn't support it, the developper can use `unsafe_get()`
* We should strive for good error messages in the general case. It's fine to have bad/no error messages in the other situations.
* When we have SSO and can control the size of the internal String buffer, we can make it big enough in this precise code path to make sure that the String bytes is always allocated on the stack (buffer) and never on the heap. When this is done, the platforms where heap allocation isn't possible will be able to use the safe method to get an item by index.

Reply:
Okay, thanks for clarifying, I'm glad you already had this discussion. I'm okay with the goal here, but it might be risky to merge this since it _will_ introduce overhead for existing (internal) users. I can push this forward, but it would be great if you could fast follow with the implementation of an unsafe accessors.

Reply:
I'll add it in this PR instead. Let's not take unnecessary risks with internal breakages.

Reply:
@laszlokindrat  I added unsafe_get to have fast access :)

Reply:
amazing, thanks!